lib/broadway_cloud_pub_sub/google_api_client.ex

```diff
@@ -58,27 +55,25 @@ defmodule BroadwayCloudPubSub.GoogleApiClient do
   defp should_retry?(_other), do: false
 
   @impl Client
-  def prepare_to_connect(module, opts) do
-    pool_name = Module.concat(module, ConnectionPool)
-
-    pool_opts =
-      opts
-      |> Keyword.get(:pool_opts, [])
-      |> Keyword.put_new_lazy(:max_connections, fn -> opts[:pool_size] end)
-
-    pool_spec = :hackney_pool.child_spec(pool_name, pool_opts)
-
-    {[pool_spec], Keyword.put(opts, :__connection_pool__, pool_name)}
+  def prepare_to_connect(name, producer_opts) do
+    finch_name = Module.concat(name, Finch)
+
+    producer_opts =
+      producer_opts
+      |> Keyword.put_new(:__internal_tesla_adapter__, FinchAdapter)
+      |> Keyword.put_new(:__internal_tesla_adapter_opts__,
+        name: finch_name,
+        receive_timeout: :infinity
+      )
+
+    {[{Finch, name: finch_name}], producer_opts}
```

Comment:
Should we allow the user to configure the pool size and count for Finch?

Reply:
Probably yes, if we go with finch

Reply:
To be clear, I think you MUST declare the pools, based on the hosts, otherwise it won't even work.

Reply:
fyi it is possible to just do: Finch.start_link(name: MyFinch) and Finch will start with a default http1 lazy pool with default sizes. Agreed about making it cinfigurable of course.

Reply:
@josevalim afaik finch starts pool on demand for the hosts, but I didn't use it for real yet, so I may be wrong.